### PR TITLE
[ENG-4583] Add analytics to new search interface

### PR DESCRIPTION
**Pull request by chth0n1x**
-   Ticket: https://openscience.atlassian.net/browse/ENG-4583
-   Feature branch: feature/search-analytics

## Purpose

The purpose of these changes is to add analytics events to the new search component.

## Summary of Changes

Data analytics name and scope properties were added to the search component to help us understand how users are utilizing the new search feature. Search analytics scope now comprise the heading, left and main layout sectors of the OSF Layout component. Correspondingly, data analytics names have been added to button, link, and input elements to determine what is being interacted with on the page.

## Screenshot(s)

Search OSF

Figure 1: Analytics event for search input

Search OSF
Figure 2: Analytics event for search button

Search OSF

Figure 3: Analytics event for start search help button

Search OSf

Search OSF

Figure 4, 5: Analytics event for next search help button

Search OSF

Figure 6: Analytics event for complete search help button

Search OSF

Figure 7: Analytics event for toggle facet filter in layout left 

Pasted Graphic 13

Figure 8: Analytics event for see more facet filters in layout left

Pasted Graphic 14

Figure 9: Analytics event for cancel facet filter in osf dialog modal

Pasted Graphic 15

Figure 10: Analytics event for apply facet filter in osf dialog modal

## Side Effects

There should not be any side effects but multiple functions being called with the queue helper in the search component handlebars file would be good to test.

## QA Notes

-Are you able to view the debug logs? 
-Are you able to extract useful data from the statistics reported? 
-Should any other be fields accounted for?

### CODE REVIEW

File: lib/osf-components/addon/components/search-page/component.ts [hunk 1]
```diff
@@ -156,4 +158,14 @@ export default class SearchPage extends Component<SearchArgs> {
...
+
+    // analyticsEvent() logs category, actionName, label, and search query
+    @action
+    analyticsEvent(actionName: string) {
+        const q = this.searchText;
+        const category = this.resourceType;
+        if (q) {
+            this.analytics.track(category, actionName, 'search-page', q);
+        }
+    }
```
Comment (adlius) [inline lib/osf-components/addon/components/search-page/component.ts:170]: I am not sure we want to actually track what users are searching.

**Review by adlius — CHANGES_REQUESTED** [review]
I think the purpose of this ticket is to add `data-analytics-name` and/or `data-analytics-scope` properties to elements, instead of tracking what our users are actually searching.

File: lib/osf-components/addon/components/search-page/component.ts (hunk 1 above)
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/component.ts]: Following our meeting, this analytics service has been removed in favor of the more modern data analytics scope and name approach.

**Comment by chth0n1x** [comment]
> I think the purpose of this ticket is to add `data-analytics-name` and/or `data-analytics-scope` properties to elements, instead of tracking what our users are actually searching.

Hi @adlius the analytics event has now been replaced with the data analytics name and scope properties. Please take a look and let me know if you have any questions.

**Comment by chth0n1x** [comment]
This PR's branch has been rebased onto feature/search-improvements as requested. Care should be taken when looking at most recent updates to the files lib/osf-components/addon/components/search-page/filter-facet/template.hbs and lib/osf-components/addon/components/search-page/template.hbs for any differences between the closed branch and the newer one.

**Comment by adlius** [comment]
It looks like the changes made to `search-page/template.hbs` is still based on the old version of the component. It has since been refactored to clean up some of the template logic.

File: lib/osf-components/addon/components/search-page/template.hbs [hunk 2]
```diff
@@ -223,121 +226,125 @@ as |layout|>
...
-                        local-class='next-button'
-                        @type='primary'
-                        onclick={{queue
-                            (fn (mut this.showTooltip1) false)
-                            (fn (mut this.showTooltip2) true)
-                        }}
-                    >
-                        {{t 'search.search-help.footer.next'}}
-                    </Button>
+<div data-analytics-scope='Search help'>
```
Comment (adlius) [inline lib/osf-components/addon/components/search-page/template.hbs:229]: Is this div necessary?

**Review by adlius — APPROVED** [review]

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**